core/lib/spree/core/error_handler/base.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Spree
+  module Core
+    module ErrorHandler
+      ##
+      # Abstract class only meant to be inherited from
+      #
+      # @example Create a custom error handler
+      #   class CustomErrorHandler < Spree::Core::ErrorHandler::Base
+      #     # Output to as debug by default
+      #     def handle(error, severity: :debug)
+      #       Rollbar.send(severity, error)
+      #       Rails.logger.send(severity, error)
+      #     end
+      #   end
+      #
+      class Base
+        class << self
+          ##
+          # Generic error handling
+          # @param error [StandardError] The error you want to handle.
+          # @param serverity [Symbol, String] The severity (i.e. debug, info, warn, error, fatal)
+          #
+          def handle(_error, _severity = :error)
+            raise NoMethodError
+          end
```

Comment:
I'd think that we might want to think about the exact sequence and type of parameters, taking into consideration reporting to external services, such as Honeybadger:

https://docs.honeybadger.io/ruby/getting-started/reporting-errors.html#passing-additional-options-to-honeybadger-notify

for example, by leaving out 

```ruby
message = error.respond_to?(:message) ? error.message : error
```

One can't do 

```ruby
ErrorHandler.report("ma mia")
```

which of course might be desirable, not sure..

Also, `severity` sounds like just one possible option, which an error can have. For example Honeybadger supports all of those:

![image](https://user-images.githubusercontent.com/1651750/43636093-0793c7ea-9709-11e8-91b9-e4273b651b76.png)

And lastly, `context` in the context of Honeybadger is an object with all sorts of additional information, can be for example 

```ruby
{ user_id: 99 }
```

Which usually comes useful to be able to pass as well, either as part of the above `options`, or maybe as an explicit parameter being so useful, as in the example that I suggested:


```ruby
class ErrorHandler
  class << self
    def report(error, context = {}, options = {})
      Honeybadger.notify(error, options.merge(context: context))

      severity = options[:severity] || :error
      message = error.respond_to?(:message) ? error.message : error
      Rails.logger.send(severity, message)
    end
  end
end
```

---

I'm just mentioning those as considerations, feel free to do without them if not deeming necessary at this point.

Reply:
We need to be careful not to build this with one error reporter in mind.  The implementation does allow overriding if needed. Also:

```ruby
ErrorHandler.report("ma mia")
# and
error = StandardError.new("ma mia")
ErrorHandler.report(error)
```
Will have the same result when it comes to the `Logger` class

There is no need to explicitly perform
```ruby
message = error.respond_to?(:message) ? error.message : error
```

Reply:
btw, thank you for your input. This is great stuff